### PR TITLE
bpo-38073: Make pwd module PEP-384 compatible

### PR DESCRIPTION
**Pull request by DinoV**
Makes the pwd module PEP-384 compatible

https://bugs.python.org/issue38073

Automerge-Triggered-By: @tiran

### CODE REVIEW

File: Misc/NEWS.d/next/Core and Builtins/2019-09-09-15-17-58.bpo-38073.ZoKYOU.rst [hunk 1]
```diff
@@ -0,0 +1 @@
+pwd module is now PEP-384 compatible
```
Comment (matrixise) [inline Misc/NEWS.d/next/Core and Builtins/2019-09-09-15-17-58.bpo-38073.ZoKYOU.rst]:
```
`:mod:`pwd` is :pep:`384` compatible
```

File: Modules/pwdmodule.c [hunk 2]
```diff
@@ -47,8 +47,13 @@ The uid and gid items are integers, all others are strings. An\n\
 exception is raised if the entry asked for cannot be found.");
 
 
-static int initialized;
-static PyTypeObject StructPwdType;
+typedef struct {
+  PyTypeObject *StructPwdType;
```
Comment (matrixise) [inline Modules/pwdmodule.c]: indent of 4?

File: Modules/pwdmodule.c [hunk 3]
```diff
@@ -310,34 +315,47 @@ static PyMethodDef pwd_methods[] = {
...
-        if (PyStructSequence_InitType2(&StructPwdType,
-                                       &struct_pwd_type_desc) < 0)
-            return NULL;
-        initialized = 1;
-    }
-    Py_INCREF((PyObject *) &StructPwdType);
-    PyModule_AddObject(m, "struct_passwd", (PyObject *) &StructPwdType);
+    pwdmodulestate *state = PyModule_GetState(m);
+    state->StructPwdType = PyStructSequence_NewType(&struct_pwd_type_desc);
+    if (state->StructPwdType == NULL)
```
Comment (matrixise) [inline Modules/pwdmodule.c]:
```c
if (state->StructPwdType == NULL) {
    return NULL
}
```

**Review by matrixise — CHANGES_REQUESTED** [review]

**Comment by DinoV** [comment]
I have made the requested changes; please review again.

**Review by tiran — APPROVED** [review]
LGTM, thanks!

File: Misc/NEWS.d/next/Core and Builtins/2019-09-09-15-17-58.bpo-38073.ZoKYOU.rst [hunk 4]
```diff
@@ -0,0 +1 @@
+`:mod:`pwd` is :pep:`384` compatible
```
Comment (tiran) [inline Misc/NEWS.d/next/Core and Builtins/2019-09-09-15-17-58.bpo-38073.ZoKYOU.rst]: sphinx doesn't like the reference here, please change the text to ``Make pwd extension module PEP-384 compatible``

**Review by tiran — CHANGES_REQUESTED** [review]

**Review by tiran — APPROVED** [review]

**Pull request merged**